### PR TITLE
Files: Episode/Network Audios & Episode/Podcast/Network Images

### PR DESCRIPTION
**Pull request by eteubert**
Not done yet, but would like a second pair of eyes on the current implementation/ideas.

I see audio files as "attachments" to either an episode or network now, implemented via polymorphic Ecto table, not unlike the Permission system. See `lib/radiator/media/audio_file_upload.ex` for public API.

Images can probably be treated in a simpler way, as it's always one image per entity, so it can be a column there (episode cover, podcast cover, network cover etc.). Rough implementation already here for user avatars.

I also wonder if the attachment system should be even more generic and be capable to take any kind of file -- but that would also make it more complex and I'm not sure it's required.

### CODE REVIEW

File: priv/repo/migrations/20190420102616_create_audio.exs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+defmodule Radiator.Repo.Migrations.CreateAudio do
+  use Ecto.Migration
```
Comment (monkeydom) [inline priv/repo/migrations/20190420102616_create_audio.exs]: Maybe we should call this AudioFile to keep the door open for an Audio that is more of the meta object we want in the future. E.g. an Audio represents a podcast episode. It consists of multiple source audio files for the tracks, and multiple processed, download ready files (e.g. the results of something coming back from auphonic), maybe in multiple versions with different inserts, etc.

File: lib/radiator/auth/user.ex [hunk 2]
```diff
@@ -33,9 +35,10 @@ defmodule Radiator.Auth.User do
   end
 
   @doc false
-  def changeset(%User{} = user, attrs) do
+  def changeset(%User{} = user, attrs \\ :invalid) do
```
[the remark this answers is not in the document]
Comment (eteubert) [reply lib/radiator/auth/user.ex]: Will remove it, I believe I copied it from some example code but I'm sure it's not needed.

File: priv/repo/migrations/20190319092448_add_users_table.exs [hunk 3]
```diff
@@ -6,6 +6,7 @@ defmodule Radiator.Repo.Migrations.AddUsersTable do
       add :name, :string
       add :email, :string
       add :display_name, :string
+      add :avatar, :string
```
[the remark this answers is not in the document]
Comment (eteubert) [reply priv/repo/migrations/20190319092448_add_users_table.exs]: No that's fine, it contains the [`arc_ecto` file reference](https://github.com/stavro/arc_ecto#add-a-string-column-to-your-schema).

File: priv/repo/migrations/20190420102616_create_audio.exs (hunk 1 above)
Comment (eteubert) [reply priv/repo/migrations/20190420102616_create_audio.exs]: Gute Idee, ist umgesetzt

**Comment by eteubert** [comment]
Travis CI currently fails because some tests require a running s3/minio instance. Apart from that this PR is complete, I think.

**Pull request merged**